gremlin-client/src/structure/value.rs

```diff
@@ -400,11 +433,18 @@ macro_rules! impl_try_from_option {
                 if let GValue::Null = value {
                     return Ok(None);
                 }
-                let res = value.try_into()?;
-                Ok(res)
+                let res: $t = value.try_into()?;
+                Ok(Some(res))
             }
         }
     };
 }
 
 impl_try_from_option!(String);
+impl_try_from_option!(i32);
```

Comment:
Ideally we'd be able to invoke this macro with some kind of loop over invoking the macro? I'm going to be repeating all these value types again for #127 I believe, so if there's a way to do a loop let me know.